plugins/elantp/fu-elantp-i2c-device.c

```diff
@@ -269,7 +365,19 @@ fu_elantp_i2c_device_setup(FuDevice *device, GError **error)
 	/* define the extra instance IDs (ic_type + module_id) */
 	fu_device_add_instance_u8(device, "ICTYPE", ic_type);
 	fu_device_build_instance_id(device, NULL, "ELANTP", "ICTYPE", NULL);
-	fu_device_build_instance_id(device, NULL, "ELANTP", "ICTYPE", "MOD", NULL);
+	if (fu_device_has_private_flag(device, FU_ELANTP_I2C_DEVICE_ABSOLUTE)) {
+		fu_device_add_instance_str(device, "DRIVER", "ELAN_I2C");
+		fu_device_build_instance_id(device,
+					    NULL,
+					    "ELANTP",
+					    "ICTYPE",
+					    "MOD",
+					    "DRIVER",
+					    NULL);
+	} else {
+		fu_device_add_flag(device, FWUPD_DEVICE_FLAG_NEEDS_REBOOT);
+		fu_device_build_instance_id(device, NULL, "ELANTP", "ICTYPE", "MOD", NULL);
```

Comment:
i think it's valuable to have this for both kinds, i.e. for the DRIVER one you'd just ignore the extra instance ID. If you need to target the opposite, then it's worth setting `DRIVER` to `none` or something and be explicit rather than "hiding" the original ID. With the  `fu_device_add_instance_str(device, "DRIVER", "none");` then you also don't need to worry about accidentally installing on the wrong device on an old fwupd version.

Reply:
Do you mean like this:
```
if (fu_device_has_private_flag(device, FU_ELANTP_I2C_DEVICE_ABSOLUTE)) {
	fu_device_add_instance_str(device, "DRIVER", "ELAN_I2C");
} else {
	fu_device_add_flag(device, FWUPD_DEVICE_FLAG_NEEDS_REBOOT);
	fu_device_add_instance_str(device, "DRIVER", "nonoe");
	fu_device_build_instance_id(device, NULL, "ELANTP", "ICTYPE", "MOD", NULL);
}
fu_device_build_instance_id(device, NULL, "ELANTP", "ICTYPE", "MOD", "DRIVER", NULL);
```

As far as I know, instance ID of existing HID devices are matched using "ELANTP", "ICTYPE", "MOD", so I avoid generating this instance ID on non-HID devices.

Reply:
> are matched using "ELANTP", "ICTYPE", "MOD",

Ahh okay, so maybe doing what you are doing makes sense. Sorry for the noise.

Reply:
After internal discussions, no updated firmware for HID devices has been uploaded yet.
Maybe the instance ID of **ELANNTP\ICTYPE_XX&MOD_YYYY&DRIVER_HID** can be used on the HID device, but the FuElantpHidDevice needs to be modified.
Do you recommend this change?

Reply:
> but the FuElantpHidDevice needs to be modified

I think that makes sense. Can you also update README.md too please.

Reply:
Thanks, updated as suggested.